oclint-reporters/reporters/PMDReporter.cpp

```diff
@@ -29,6 +47,29 @@ class PMDReporter : public Reporter
         }
         return output;
     }
+
+    void writeViolation(std::ostream &out,
+                        const Violation &violation,
+                        const PMDRuleDescription *ruleDescription)
+    {
+        out << "<file name=\"" << violation.path << "\">" << std::endl;
+        out << "<violation ";
+        out << "begincolumn=\"" << violation.startColumn << "\" ";
+        out << "endcolumn=\"" << violation.endColumn << "\" ";
+        out << "beginline=\"" << violation.startLine << "\" ";
+        out << "endline=\"" << violation.endLine << "\" ";
+        if (ruleDescription)
+        {
+            out << "priority=\"" << ruleDescription->priority << "\" ";
```

Comment:
Wondering if it's a good idea to add a method `writeViolation(std::ostream &out)` in `PMDRuleDescription` class, which contains this line and below two lines. And here, we can basically have `ruleDescription.writeViolation(out)`.